spec/system/distribution_system_spec.rb

```diff
@@ -0,0 +1,235 @@
+RSpec.describe "Distributions", type: :system do
+  before do
+    sign_in(@user)
+    @url_prefix = "/#{@organization.to_param}"
+
+    @partner = create(:partner, organization: @organization)
+    # allow_any_instance_of(StorageLocation).to receive(:geocode).and_return(true)
+    @storage_location = create(:storage_location, organization: @organization)
+    setup_storage_location(@storage_location)
+  end
+
+  it "User creates a new distribution" do
+    with_features email_active: true do
+      visit @url_prefix + "/distributions/new"
+
+      select @partner.name, from: "Partner"
+      select @storage_location.name, from: "From storage location"
+
+      fill_in "Comment", with: "Take my wipes... please"
+
+      expect do
+        click_button "Save", match: :first
+      end.to change { PartnerMailerJob.jobs.size }.by(1)
+
+      expect(page).to have_content "Distributions"
+      expect(page.find(".alert-info")).to have_content "reated"
+    end
+  end
+
+  it "User doesn't fill storage_location" do
+    visit @url_prefix + "/distributions/new"
+
+    select @partner.name, from: "Partner"
+    select "", from: "From storage location"
+
+    click_button "Save", match: :first
+    expect(page).to have_content "An error occurred, try again?"
+  end
+
+  context "With an existing distribution" do
+    let!(:distribution) { create(:distribution, :with_items, agency_rep: "A Person", organization: @user.organization) }
+
+    before do
+      visit @url_prefix + "/distributions"
+    end
+
+    it "the user can make changes to it" do
+      click_on "Edit", match: :first
+      expect do
+        fill_in "Agency representative", with: "SOMETHING DIFFERENT"
+        click_on "Save", match: :first
+        distribution.reload
+      end.to change { distribution.agency_rep }.to("SOMETHING DIFFERENT")
+    end
+
+    it "the user can reclaim it" do
+      expect do
+        accept_confirm do
+          click_on "Reclaim"
+        end
+        page.find ".alert"
```

Comment:
@armahillo one difference from the Feature spec. This was failing originally (`expected Distribution.count to change by 1, but did not change`), and on a whim I added this line after seeing it used in the spec on line 67, which looks similar. And that fixes it, I'm guessing because it just forces Capybara to wait a little longer before running the change expectation?

Is this OK? Is there a better way I could handle it?

Reply:
@pcrglennon Sorry for missing this! It's been a flurry of traffic since Rails conf and this got lost in the shuffle.

`page.find {...}` is an effective way to introduce a blocking-state in Capybara; it's the preferred way of waiting for the document to change. So yes, this is great! Nice work. :)

Reply:
No worries! And thanks